### DISCUSSION TITLE
emscripten failed to load in iOS Safari\Chrome if Worker Threads are greater than 10.  (macOS is OK)

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#20377] [Q&A]
My emscripten app is failed to load in **iOS 16\17 Safari\Chrome** if Worker Threads are greater than 10 (macOS is OK).
I would be glad if anybody knows about any workaround for this.
here are the emscripten flags that I use to build my App:
`-s ALLOW_TABLE_GROWTH -s WASM_BIGINT  -s EXPORTED_RUNTIME_METHODS=ccall,cwrap,addFunction  -s EXPORTED_FUNCTIONS=_main  -g  -gsplit-dwarf -gdwarf-5 -gpubnames --bind -s USE_PTHREADS -s ALLOW_MEMORY_GROWTH=1    -s ASSERTIONS  -s FETCH   -s NO_DISABLE_EXCEPTION_CATCHING -s TEXTDECODER=0  -lidbfs.js -s PTHREAD_POOL_SIZE=15 -s ERROR_ON_WASM_CHANGES_AFTER_LINK -s REVERSE_DEPS=all -s STACK_SIZE=1mb -s DEFAULT_PTHREAD_STACK_SIZE=1mb -s ALLOW_BLOCKING_ON_MAIN_THREAD=1  -s PTHREAD_POOL_SIZE_STRICT=2 
`

**Comment by aharondavid** [comment]
More info: when I compiled with Optimizations Flags,  then I can raise up a 20 Worker Threads, I need to raise up at least 40 Worker Threads.
Trying to decrease Memory Consuming doesn't help. it is not seems a Memory problem.
There is no limitation on the number of Worker Threads via a simple html JS. so I wander  
What is special about Worker Threads that are raised up by emscripten pthreads.
@sbc100, do you have any clue ?

**Reply by sbc100** [reply]
Each of emscripten's threads needs to independently instantiate the module so that memory using of each thread will be proportional to the size of your module.   The fact that building with optimization enabled lets you start more threads points to this been memory/resource issue.   Do you have some kind remove debug method by which you can monitor the resources used by a web page?  I assume safari has something like that?

**Comment by aharondavid** [comment]
Hi, glad to hear you :)

The WASM and its DATA are about 50M, so it is might explain a Memory issue.
see my question about that here:
https://github.com/emscripten-core/emscripten/discussions/20458

**Comment by aharondavid** [comment]
In iOS (version 16) Safari\Chrome It seems that there is no limitation on number of Worker Threads, but there is limitation on memory size - the max size is 1.5G .
